cni/pkg/install/install.go

```diff
@@ -153,9 +154,21 @@ func (in *Installer) Cleanup() error {
 // sleepCheckInstall verifies the configuration then blocks until an invalid configuration is detected, and return nil.
 // If an error occurs or context is canceled, the function will return the error.
 // Returning from this function will set the pod to "NotReady".
-func (in *Installer) sleepCheckInstall(ctx context.Context) error {
-	targets := append(slices.Clone(in.cfg.CNIBinTargetDirs), in.cfg.MountedCNINetDir, constants.ServiceAccountPath)
-
+func (in *Installer) sleepCheckInstall(ctx context.Context, installedBinFiles []string) error {
+	// Watch our specific binaries, in each configured binary dir.
+	// We may or may not be the only CNI plugin in play, and if we are not
+	// we shouldn't fire events for binaries that are not ours.
+	var binPaths []string
+	for _, bindir := range in.cfg.CNIBinTargetDirs {
+		for _, binary := range installedBinFiles {
+			binPaths = append(binPaths, filepath.Join(bindir, binary))
```

Comment:
This may only apply to mounts which k8s handles by symlinks, but you might need to watch the folder rather than the actual binary. You can filter changes in the folder to only those that matched to get the same effect?

Might not be needed though, not sure how good our test coverage of this is to check

Reply:
> // Watching individual files (rather than directories) is generally not
> // recommended as many tools update files atomically. Instead of "just" writing
> // to the file a temporary file will be written to first, and if successful the
> // temporary file is moved to to destination removing the original, or some
> // variant thereof. The watcher on the original file is now lost, as it no
> // longer exists.

I think our solution may work by the fact that one any change, we stop all the watches anyways. Still feels a bit fragile though

Reply:
```
This may only apply to mounts which k8s handles by symlinks, but you might need to watch the folder rather than the actual binary. You can filter changes in the folder to only those that matched to get the same effect?
```
Yep that would also work.

I missed that  blurb in `fsnotify` - I'll change the logic accordingly.